lib/MySQL_HostGroups_Manager.cpp

```diff
@@ -4499,6 +4556,214 @@ void MySQL_HostGroups_Manager::read_only_action(char *hostname, int port, int re
+		mydb->execute("DELETE FROM mysql_servers");
+		generate_mysql_servers_table();
+
+		//if (GloAdmin && GloAdmin->checksum_variables.checksum_mysql_servers) 
+		{
+			char* error = NULL;
+			int cols = 0;
+			int affected_rows = 0;
+			SQLite3_result* resultset = NULL;
+			mydb->execute_statement(MYHGM_GEN_ADMIN_RUNTIME_SERVERS, &error, &cols, &affected_rows, &resultset);
+			save_runtime_mysql_servers(resultset); // assigning runtime_mysql_servers with updated mysql server resultset 
+			
+			resultset = NULL;
+
+			char* query = (char*)"SELECT hostgroup_id, hostname, port, gtid_port, CASE status WHEN 0 OR 1 OR 4 THEN 0 ELSE status END status, weight, compression, max_connections, max_replication_lag, use_ssl, max_latency_ms, comment FROM mysql_servers WHERE status<>3 ORDER BY hostgroup_id, hostname, port";
+			mydb->execute_statement(query, &error, &cols, &affected_rows, &resultset);
+
+			if (resultset) {
+				if (resultset->rows_count) {
+					table_resultset_checksum[HGM_TABLES::MYSQL_SERVERS] = resultset->raw_checksum();
+
+					hgsm_mysql_servers_checksum = table_resultset_checksum[HGM_TABLES::MYSQL_SERVERS];
+				}
+				delete resultset;
+			} else {
+				proxy_info("Checksum for table %s is 0x%lX\n", "mysql_servers", (long unsigned int)0);
+			}
+
+			uint64_t hash = 0, hash2 = 0;
+			SpookyHash myhash;
+
+			myhash.Init(19, 3);
+			hash = table_resultset_checksum[HGM_TABLES::MYSQL_SERVERS];
+			myhash.Update(&hash, sizeof(hash));
+
+			hash = table_resultset_checksum[HGM_TABLES::MYSQL_REPLICATION_HOSTGROUPS];
+			myhash.Update(&hash, sizeof(hash));
+
+			hash = table_resultset_checksum[HGM_TABLES::MYSQL_GROUP_REPLICATION_HOSTGROUPS];
+			myhash.Update(&hash, sizeof(hash));
+
+			hash = table_resultset_checksum[HGM_TABLES::MYSQL_GALERA_HOSTGROUPS];
+			myhash.Update(&hash, sizeof(hash));
+
+			hash = table_resultset_checksum[HGM_TABLES::MYSQL_AWS_AURORA_HOSTGROUPS];
+			myhash.Update(&hash, sizeof(hash));
+
+			hash = table_resultset_checksum[HGM_TABLES::MYSQL_HOSTGROUP_ATTRIBUTES];
+			myhash.Update(&hash, sizeof(hash));
+
+			myhash.Final(&hash, &hash2);
+
+			char buf[80];
+			uint32_t d32[2];
+			memcpy(&d32, &hash, sizeof(hash));
+			sprintf(buf, "0x%0X%0X", d32[0], d32[1]);
+			pthread_mutex_lock(&GloVars.checksum_mutex);
+			GloVars.checksums_values.mysql_servers.set_checksum(buf);
+			GloVars.checksums_values.mysql_servers.version++;
+			//struct timespec ts;
+			//clock_gettime(CLOCK_REALTIME, &ts);
+			time_t t = time(NULL);
+
+			GloVars.checksums_values.mysql_servers.epoch = t;
+
+			GloVars.checksums_values.updates_cnt++;
+			GloVars.generate_global_checksum();
+			GloVars.epoch_version = t;
+			pthread_mutex_unlock(&GloVars.checksum_mutex);
+		}
+	}
+	wrunlock();
+	unsigned long long curtime2 = monotonic_time();
+	curtime1 = curtime1 / 1000;
+	curtime2 = curtime2 / 1000;
+	proxy_info("MySQL_HostGroups_Manager::read_only_action_v2() locked for %llums (server count:%d)\n", curtime2 - curtime1, mysql_servers.size());
```

Comment:
This will be printed after each `read_only_action`, this is way to much verbosity for the error log. 
This probably belongs in a `proxy_debug` for `PROXY_DEBUG_MONITOR` with high verbosity, for example, `7`.

Reply:
Done